### PR TITLE
CASMINST-5851

### PR DESCRIPTION
**Pull request by atifsyedali**
## Summary and Scope

Finally fixes [CASMINST-5851](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-5851)

The previous fix didn't work on lemondrop.

I hot-patched a new fix on lemondrop and let it run for a few days. It seems the new fix that was hot-patched has not produced any duplicate workflows.

About the new fix: it tweaks the timeout value for metacontroller from 10s to 30s. This gives enough time for the first call to complete on large stages such as deliver-product.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMINST-5851](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-5851)

## Testing

lemondrop

### Test description:

Ran `IUF run` multiple times and didn't notice any duplicate workflows.

## Risks and Mitigations

No

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: src/api/controllers/v1/iuf/sessions.go [hunk 1]
```diff
@@ -212,20 +196,48 @@ func (u IufController) Sync(context *gin.Context) {
...
 
 		session.CurrentState = iuf.SessionStateDebug
 		err = u.iufService.UpdateSessionAndActivity(session, fmt.Sprintf("Unknown state %s", session.CurrentState))
-		u.logger.Error(err)
+		if err != nil {
+			context.JSON(500, utils.ResponseError{Message: err.Error()})
+			return
+		}
 
+		context.JSON(500, utils.ResponseError{Message: err.Error()})
```
Comment (haasken-hpe) [inline src/api/controllers/v1/iuf/sessions.go:212]: is this line a mistake? won't `err` be `nil` here?

File: src/api/controllers/v1/iuf/sessions.go [hunk 2]
```diff
@@ -212,20 +196,48 @@ func (u IufController) Sync(context *gin.Context) {
...
 		return
 	}
+
+	// why did we end up here? Golang really needs better static analysis.
+	context.JSON(500, utils.ResponseError{Message: "Sync: Unknown code path. Should't have landed here."})
```
Comment (haasken-hpe) [inline src/api/controllers/v1/iuf/sessions.go:217]:
```suggestion
	context.JSON(500, utils.ResponseError{Message: "Sync: Unknown code path. Shouldn't have landed here."})
```

**Review by haasken-hpe — APPROVED** [review]
Looks like a possible NPE was introduced. Otherwise, this seems fine to me.

File: src/api/controllers/v1/iuf/sessions.go (hunk 1 above)
Comment (atifsyedali) [reply src/api/controllers/v1/iuf/sessions.go:212]: Oh right good catch.

**Pull request merged**